### PR TITLE
[FS-180]: List Of Projects API Integration

### PR DESCRIPTION
**Pull request by erfan-goodarzi**
This PR lists Project API integration.

### CODE REVIEW

File: src/components/CreateProject/ProjectForm/CreateProjectForm.tsx [hunk 1]
```diff
@@ -28,7 +28,22 @@ const FormSchema = yup
...
 
 export const CreateProjectForm = ({ dismiss }: Props) => {
-  const [createProject, { loading }] = useCreateProjectMutation();
+  const [createProject, { loading }] = useCreateProjectMutation({
+    update(cache, { data }) {
+      const newProject = data?.insert_project_one
+      const prevProjects = cache.readQuery<ProjectListQuery>({ query: ProjectListDocument })
+
+      if (prevProjects && newProject) {
+        console.log(prevProjects, newProject)
```
Comment (AmirabbasJ) [inline src/components/CreateProject/ProjectForm/CreateProjectForm.tsx:37]: we don't need the log

File: src/components/ProjectList/ProjectTable/ProjectTable.tsx [hunk 2]
```diff
@@ -1,35 +1,27 @@
 import { messages } from '@camp/messages';
 import { createTestAttr } from '@camp/utils';
 import { Table, useMantineTheme } from '@mantine/core';
+import { useProjectListQuery } from '../../../data-layer/operations/__generated__/typesAndHooks';
```
Comment (AmirabbasJ) [inline src/components/ProjectList/ProjectTable/ProjectTable.tsx:4]: you should export `useProjectListQuery` in data-layer and please import it from `'@camp/data-layer'`

File: src/components/ProjectList/ProjectTable/ProjectTable.tsx [hunk 3]
```diff
@@ -1,35 +1,27 @@
...
-    id: '3',
-    name: 'خرید مک بوک برای تیم فول استکس',
-  },
-];
-
 export const ProjectTable = () => {
   const { colors } = useMantineTheme();
   const t = messages.projects.list.table.columns;
   const columns = t.map(column => <th key={column}>{column}</th>);
+  const { data, loading, error, refetch } = useProjectListQuery()
```
Comment (AmirabbasJ) [inline src/components/ProjectList/ProjectTable/ProjectTable.tsx:14]: refetch is unused you can remove it

File: src/components/ProjectList/ProjectTable/ProjectTable.tsx [hunk 4]
```diff
@@ -6,32 +7,18 @@ import { ProjectTableRow } from './ProjectTableRow';
...
-
 export const ProjectTable = () => {
   const { colors } = useMantineTheme();
   const t = messages.projects.list.table.columns;
   const columns = t.map(column => <th key={column}>{column}</th>);
+  const { data, loading, error } = useProjectListQuery();
+
+  if (error) console.log(error);
+
+  if (loading) return <h1>loading...</h1>;
```
Comment (erfan-goodarzi) [inline src/components/ProjectList/ProjectTable/ProjectTable.tsx:18]: U need to use `<FullPageLoader/>` component here.

File: src/components/CreateProject/ProjectForm/CreateProjectForm.tsx [hunk 5]
```diff
@@ -28,7 +32,25 @@ const FormSchema = yup
...
+        cache.writeQuery({
+          query: ProjectListDocument,
+          data: {
+            project_aggregate: {
+              nodes: [...prevProjects.project_aggregate.nodes, newProject],
+            },
+          },
+        });
+      }
+    },
```
Comment (ASafaeirad) [inline src/components/CreateProject/ProjectForm/CreateProjectForm.tsx:52]: This is the data layer's concern, let's keep it for now but we should move it there later.

File: src/components/ProjectList/ProjectTable/ProjectTable.tsx [hunk 6]
```diff
@@ -1,37 +1,25 @@
...
-  },
-];
-
 export const ProjectTable = () => {
   const { colors } = useMantineTheme();
   const t = messages.projects.list.table.columns;
   const columns = t.map(column => <th key={column}>{column}</th>);
+  const { data, loading, error } = useProjectListQuery();
+
+  if (error) console.log(error);
```
Comment (ASafaeirad) [inline src/components/ProjectList/ProjectTable/ProjectTable.tsx:17]: It's better to return a JSX here to not continue.

**Review by ASafaeirad — CHANGES_REQUESTED** [review]

**Review by ASafaeirad — APPROVED** [review]

**Pull request merged**